src/main/kotlin/no/nav/familie/ef/sak/service/steg/StegService.kt

```diff
@@ -29,6 +30,14 @@ class StegService(
         }
     }
 
+    @Transactional
+    fun håndterInngangsvilkår(behandling: Behandling): Behandling {
```

Comment:
@charliemidtlyng etter å ha sett på steg-koden så vet jag ikke riktigt om jeg skjønner opplegget.
Hva er ønskelig her, når vi skal vurdere inngångsvilkår så tenkte jeg att steget på behandlingen burde være `VILKÅRSVURDERE_INNGANGSVILKÅR`, men det virker ikke helt med koden, fordi `InngangsvilkårSteg` har ju stegtype `VILKÅRSVURDERE_INNGANGSVILKÅR` og i stegservice gjøres det en sjekk att behandlingsteg.stegType er etter behandlingen sin stegtype.. 

Så er det forventet att behandlingen har steget før ingangsvilkår til att inngangsvilkåret er validert?

Reply:
Behandlingen starter på:
REGISTRERE_OPPLYSNINGER.

Når du utfører inngangsvilkår medlemskap så blir denne kjørt.  Da har behandlingen  `REGISTRERE_OPPLYSNINGER` og input er `VILKÅRSVURDERE_INNGANGSVILKÅR`. Følgelig blir den nye stegtypen på behandlingen `VILKÅRSVURDERE_INNGANGSVILKÅR` når den blir validert og godkjent. 

Men du har jo et poeng - må jo kunne gå "tilbake", ja.

Reply:
Det høres for meg litt rart ut. Tenker det er mer logisk hvis behandlingen er i steg `VILKÅRSVURDERE_INNGANGSVILKÅR` og må validere inngangsvilkår for å gå videre til neste steg

Du er ju egentlige på steg 2 og ønsker å gå videre til steg 3, då burde det være naturlig å validere allt for steg 2 før man er på steg 3?

Reply:
Tenker, hvis man ønsker å åpne en oppgave i frontend og komme direkt til det steget, eks inngangsvilkår for å vurdere inngangsvilkåret, så er det ju enklest hvis steget på behandlingen er inngangsvilkår og ikke REGISTRERE_OPPLYSNINGER

Reply:
Her er flyten slik den er lagt opp i denne PR-en: 
Dummy-endepunktet oppretter behandling. Behandlingen starter på steget `REGISTRERE_OPPLYSNINGER`. I samme metode utfører vi dette steget, og behandlingen settes i steget `VILKÅRSVURDERE_INNGANGSVIKÅR`. Den står i dette steget helt til man kaller validerings-endepunktet i `VurderingController` som vil utføre steget. Hvis alle inngangsvilkårene er vurdert, settes behandlingen i steg `VILKÅRSVURDERE_STØNAD`. 

Det som kanskje kan virke litt ulogisk med BehandlingSteg-rammeverket er at tidspunktet du setter behandlingen i et steg og tidspunktet du faktisk utfører det steget er helt frikobla fra hverandre. Jeg tenker på steg-feltet i behandlingen som "det neste som skal skje i saksbehandlingsløypa". Siden disse to tingene er helt frikobla og utførelsen av et steg kan trigges fra hvor som helst, må vi ha en slags sikkerhetsmekanisme for å unngå at man utfører et steg som behandlingen ikke er klar for. For eksempel gir det ikke noe mening å utføre `SEND_TIL_BESLUTTER` hvis behandlingen står i steget `VILKÅRSVURDERE_INNGANGSVILKÅR`. Det er dette `stegType.kommerEtter(behandling.steg)` gjør for oss. Det er likevel fullt mulig å gå tilbake og utføre feks `VILKÅRSVURDERE_INNGANGSVILKÅR` på nytt selv om behandlingen står i feks steg `VILKÅRSVURDERE_STØNAD`, og det er fullt mulig å utføre `VILKÅRSVURDERE_INNGANGSVILKÅR` når behandlingen står i dette steget. `kommerEtter`- sjekken er grunnen til at jeg måtte trigge utførelsen av `REGISTRERE_OPPLYSNINGER` i dummy-endepunktet, hvis ikke ville den nektet oss å utføre `VILKÅRSVURDERE_INNGANGSVILKÅR` når vi kaller validering-endepunktet.

Reply:
Jeg hadde missoppfattat denne helt, jeg tolket koden som att den var i statet `REGISTRERE_OPPLYSNINGER` når man kallade på validerInngangsvilkår av noen grunn.